oat-sa/extension-tao-itemqti
# Breaking/nex 91/integrate npm item runner

Jira ticket: https://oat-sa.atlassian.net/browse/NEX-91
Vendorisation of item runner (reintegration)

**What I did:**
- deleted all js files & test files which were moved to `tao-item-runner-qti-fe`
- registered new npm paths in php via install script & updater
- registered new npm paths in `paths.json` for grunt test task (extractable by tao-core helper)
- created `package.json` to install the new npm package
- enabled composer-npm-bridge to automatically install packages

**Prerequisites:**
- [x] https://github.com/oat-sa/tao-core/pull/2195
- [x] https://github.com/oat-sa/tao-item-runner-fe/pull/6
- [x] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/3
- [x] release & publish on npmjs `tao-item-runner` @ 0.3.0
- [x] release & publish on npmjs `tao-item-runner-qti` @ 0.3.0
- [x] update above references in this PR's package.json, & commit package-lock
- [x] also check the dependency to `tao` and `taoItems` in manifest.php, it may need updating...

**How to test:**
Install this version of the taoQtiItem extension in your TAO platform.
Verify that the `@oat-sa/tao-item-runner-qti` and `@oat-sa/tao-item-runner` npm packages were installed to views/node_modules.
Verify that TAO backoffice and test delivery still work with no 404 errors (DEBUG mode).
Verify that `npx grunt taoqtiitembundle` bundle generation task completes, and site works (PROD mode)
Verify that `npx grunt connect:dev taoqtiitemtest` finds and runs the remaining tests (I believe it's still necessary to create a `views/js/mathjax/MathJax.js` file for this to work)
